test-suite/TestUtils.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE DataKinds #-}
+module TestUtils
+  ( getValue
+  , getDigit
+  , getDigitBBP
+  , E
+  ) where
+
+import Data.Proxy   (Proxy)
+import Data.List    (foldl')
+import Data.Fixed   (mod', HasResolution(..), Fixed)
+
+import GHC.TypeLits (Nat, KnownNat, SomeNat(..), natVal, someNatVal)
+
+import Data.ExactPi
+
+-- E n generalises E2/E3/E6/E12 from Data.Fixed to give more precise
+-- fixed-precision arithmetic: Fixed (E 30) has 30 decimal places.
+data E (n :: Nat)
```

Comment:
By the way, this was a pretty good idea. Can it not be considered for addition to [`Data.Fixed`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Fixed.html)?

Reply:
I'll try making this change in `base` in a local copy of GHC first, to see if it's possible.

Reply:
Sure - I'm not really sure how to add it myself (and a little too intimidated) so feel free to go ahead.

Reply:
>and a little too intimidated

Haha 😄, we've all been there before, do take your time to prepare for a first contribution, but don't be intimidated, GHC's maintainers and contributors will be more than happy to review your work!

There are [these](https://ghc.haskell.org/trac/ghc/wiki/Building/Preparation) [resources](https://ghc.haskell.org/trac/ghc/wiki/Building/GettingTheSources) [available](https://ghc.haskell.org/trac/ghc/wiki/Building/QuickStart), with emphasis on [this one](https://ghc.haskell.org/trac/ghc/wiki/Newcomers) for people starting out. I'll submit a ticket for this feature, and I will not forget to give full credit to you 👍.

Reply:
It is worth to prepare a PR to https://github.com/ghc/ghc in addition to [your ticket](https://ghc.haskell.org/trac/ghc/ticket/15622). In simple cases PRs can be merged into GHC directly.

Reply:
@Bodigrim https://github.com/ghc/ghc/pull/196

Reply:
@b-mehta some information - your idea is now in [GHC's master](https://gitlab.haskell.org/ghc/ghc/commit/a31b24a591aecc734449d3af479e3a3d1834b0ed)!
In GHC 8.10.1 this will no longer be necessary.